### PR TITLE
[mdatagen] Add flag to indicate seeking new code owners

### PR DESCRIPTION
**Pull request by djaglowski**
Same implementation as https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31167

### CODE REVIEW

**Review by songy23 — APPROVED** [review]
LGTM

**Review by atoulme — APPROVED** [review]

File: cmd/mdatagen/testdata/readme_with_status_codeowners_and_seeking_new.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Some component
+
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [beta]: metrics   |
+| Distributions | [contrib] |
+| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Areceiver%2Ffoo%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Areceiver%2Ffoo) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Areceiver%2Ffoo%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Areceiver%2Ffoo) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@foo](https://www.github.com/foo) \| Seeking more code owners! |
```
Comment (astencel-sumo) [inline cmd/mdatagen/testdata/readme_with_status_codeowners_and_seeking_new.md:9]: Perhaps would be good to link to some docs on what it takes to be a code owner?

Comment (djaglowski) [reply cmd/mdatagen/testdata/readme_with_status_codeowners_and_seeking_new.md:9]: I think the best place to link is already included under the "Code Owners" string

Comment (astencel-sumo) [reply cmd/mdatagen/testdata/readme_with_status_codeowners_and_seeking_new.md:9]: Right, it's already there :smile:

**Comment by djaglowski** [comment]
cc @open-telemetry/collector-maintainers

**Review by dmitryax — APPROVED** [review]

**Pull request merged**